Bug 769457 More info re register views

goodvibes2 opened the pull request:
FYI Re screen dumps: I upgraded my debian 8 (Jessie) vm to 9 (stretch) so I could get screen dumps of the register which included the associated status flag in the double line view which was implemented in 2.6.14. Unfortunately, in the Clearlooks-phenix theme, this changed the titlebar colour from white text on a light blue background to black text on a light grey background, and Minimise, Maximise & Close icons are much smaller. I think the colour changes are related to mods for showing menu items in the window titlebar instead of the top line of the desktop. As I have previously spent days trying to get what I think are the required screen colours for screen dumps in Ubuntu, and eventually created the debian Jessie vm for just that purpose, these colours will have to do.

sunfish62 commented on guide/C/ch_txns.xml:
```diff
@@ -349,10 +349,10 @@
           transaction has a split assigned to the account. If a given transaction
-          has more than one split assigned to a single account, then in Basic and
-          Auto-Split view modes that transaction will appear in the register one
+          has more than one split assigned to the register account, then in Basic and
```
I prefer the original; "the register account" is not a term that I think is as clear or accurate as the original.

gjanssens replied:
I agree on this one. "the register account" requires me to read the sentence more than once to grasp what is meant. View modes to views is fine for me.

goodvibes2 replied:
The text, before I modified it, says to me that a transaction is shown whenever there is more than 1 split to "any" single account but it is only when that account is the account that was "opened" (or Open SubAccounts) that this occurs.
How about:
An account register displays a transaction when that transaction has a split assigned to the
account. If a given transaction has more than one split assigned to the account, then in Basic
and Auto-Split view that transaction will appear in the register one time for each split
assigned to that account.

sunfish62 replied:
Well, I guess that's one way to read it. I didn't get that. I guess your new text works, though.

goodvibes2 replied:
Done

sunfish62 commented on help/C/Help_ch_Transactions.xml:
```diff
@@ -56,14 +57,21 @@ Translators:
-    will select the desired option. When the window closes the style will revert to the Basic Ledger style. To
-    make this change permanent change the style in the <guilabel>Register Defaults</guilabel> tab of the
-    <guilabel>&app; Preferences</guilabel>.</para>
+    <para>The default view or style can be temporarily changed for the current
```
"The  default view or style" would be better as "The default view" to follow the example immediately above.
Also, "can be temporarily changed" seems misleading. I prefer "can be changed" and adding a sentence following: "The selected view will remain in effect until the register is closed and reopened."

goodvibes2 replied:
Re View or Style:
I agree, consistently using View instead of Style or Mode would be better but the Preference is called "Default Style" so I didn't change what others have already written.
I think the preference should be renamed to "Default View".
Shall I raise a bug for that?
Shall I still remove Style from here?

goodvibes2 replied:
Re: temporarily changed:  I agree.
How about:
The view of an open register can be changed by options in the View menu.
A register will always open in the Default Style set in the Register Defaults tab of the GnuCash Preferences.

sunfish62 replied:
Your change explains how the view will behave when it's opened; my language makes clear that the view will remain in effect until the register is closed. I prefer my language, since we are discussing how the open register behaves here. Yours would be better placed somewhere else, I believe.
As for raising a bug, I think it wouldn't necessarily hurt much to get the menu changed, so go for it.

goodvibes2 replied:
Sorry, I don't agree. The important thing is the opening of the register, not the closing. My text is clear and more concise. I have implemented the text I last suggested.
I have raised https://bugzilla.gnome.org/show_bug.cgi?id=780229 :
 Preferences, Register Defaults tab: "Default Style" should be "Default View"

sunfish62 replied:
Ok, so now we don't agree. What now? As I said, in this context, we are discussing how to modify the register view after it has already been opened, which to me recommends my language. Again as I said, your language is valid, but belongs elsewhere.

jralls replied:
Just rip out the whole section. It's covered much better in the Guide, section 4.2.

goodvibes2 replied:
Hi John,
What do you mean by the "whole section"?
If you mean the sect1 "Changing the Register View" that would include all my new screendumps.

jralls replied:
Yup. It's already covered, with plenty of screen shots, in the Guide. There's no sense in duplicating that here, screenshots included.

goodvibes2 replied:
Hi John,
Given Frank's recent mod to http://wiki.gnucash.org/wiki/Git_For_Newbies,
Would you prefer:
1) I amend the last commit and force push
2) add another commit to this PR
or
3) Close this PR and create another?

jralls replied:
Whatever is easiest for you.  You've done several force-pushes in the last few days. Have you had any Github problems?

goodvibes2 replied:
No problems - just want to do the right thing.

sunfish62 commented on help/C/Help_ch_Transactions.xml:
```diff
@@ -74,12 +82,13 @@ Translators:
 
       <listitem>
         <para><guimenu>Auto-Split Ledger:</guimenu> This style will automatically expand the splits in any
```
Although this predates, I think "style" should be "view" for consistency.

goodvibes2 replied:
Done

sunfish62 commented on help/C/Help_ch_Transactions.xml:
```diff
@@ -74,12 +82,13 @@ Translators:
       <listitem>
-        <para><guimenu>Transaction Journal:</guimenu> This style expands all transactions so the complete
-        transaction can be seen.</para>
+        <para><guimenu>Transaction Journal:</guimenu> This style expands all
```
"style"->"view"

goodvibes2 replied:
Done

sunfish62 commented on help/C/Help_ch_Transactions.xml:
```diff
@@ -88,12 +97,191 @@ Translators:
 
+    <sect2 id="reg-views-filter-sort">
+      <title>Filter or Sort</title>
+      <para>The two other options to change the view of the register are; to restrict displayed items with the
```
semi-colon grammatically incorrect here. Should be colon. Also, stylistically might prefer prefixing each option with a number.

goodvibes2 replied:
I didn't write that, but I agree. I shall fix.

gjanssens commented on help/C/Help_ch_Transactions.xml:
```diff
@@ -43,8 +43,9 @@ Translators:
+      <para>Importing transactions either from an <acronym>OFX</acronym>,
+        <acronym>QFX</acronym>, <acronym>QIF</acronym> or <acronym>CSV</acronym>
+        file or via "<acronym>OFX</acronym> on-line banking".</para>
     </listitem>
```
Is it your intention to be exhaustive here ? If you you'd also need to add MT940, MT942 and DTAUS as import formats. Perhaps here is suffices to mention that importing can be done from a number of supported file formats or via online banking. Which those are exactly probably belong in a chapter on importing.

goodvibes2 replied:
Hi Geert,
Re new style window frame:
Do you mean crop out the titlebar from screen images?
I think the window title is important, especially when it contains variable info, like the Register window, which includes the account in the title.
Whenever I see screenshots without the titlebar, I think they look disembodied. It doesn't overly worry me that there are different style windows in the manuals. I thought I'd mention it though. The new debian style actually looks a lot like Windows 10 now so that will feel familiar to those users, and the white background titlebar doesn't stand out like a colored background would.
On balance, I'd prefer to keep the titlebar.

Re imports: As we don't yet have good import documentation, I'll add the types you mention so at least they are listed somewhere other than the menus.

sunfish62 replied:
I would prefer this spot not to delineate ANY of the formats; it would be appropriate here to say:
"Importing transactions either from a file or via on-line banking."
The fact that there isn't good documentation on importing is not a reason to add these formats here. It seems to me that the Guide is a better place to put information about importing data, since importing involves so much more than simply pressing buttons and choosing options.

goodvibes2 replied:
I have changed it to say:
"Importing transactions either from a file or via on-line banking." as you suggest.

gjanssens commented:
On maandag 13 maart 2017 23:22:31 CET Chris Good wrote:
> goodvibes2 commented on this pull request.
> 
> > @@ -43,8 +43,9 @@ Translators:
>      </listitem>
> 
>      <listitem>
> -      <para>Importing transactions either from a
> <acronym>QIF</acronym>/<acronym>OFX</acronym> file or via -     
> "<acronym>OFX</acronym> on-line banking".</para>
> +      <para>Importing transactions either from an <acronym>OFX</acronym>,
> +        <acronym>QFX</acronym>, <acronym>QIF</acronym> or
> <acronym>CSV</acronym> +        file or via "<acronym>OFX</acronym> on-line
> banking".</para> </listitem>
> 
> Hi Geert,
> Re new style window frame:
> Do you mean crop out the titlebar from screen images?
> I think the window title is important, especially when it contains variable
> info, like the Register window, which includes the account in the title.
> Whenever I see screenshots without the titlebar, I think they look
> disembodied. It doesn't overly worry me that there are different style
> windows in the manuals. I thought I'd mention it though. The new debian
> style actually looks a lot like Windows 10 now so that will feel familiar
> to those users, and the white background titlebar doesn't stand out like a
> colored background would. On balance, I'd prefer to keep the titlebar.

That's fine with me as well.

goodvibes2 commented:
I have amended my last commit and force pushed to my remote repo.

goodvibes2 commented:
I have removed Help section "Changing the Register View"  (and fixed links to it) in a new commit and pushed to my repo. Ready for review again thanks.

sunfish62 commented:
Chris, would it be too much to add a sentence to the section before this that says:
Gnucash had several ways of displaying transactions in an account register. See <guide section> for more information about this.

sunfish62 commented:
Sorry about the typo

goodvibes2 commented:
Sorry, I've spent way too much time on this already and want to move on. Also, I don't want to possibly delay this being included in this weekends release.

jralls commented:
Merged after rebasing.
